cli/cmd/manifestGet.go

```diff
@@ -1,39 +1,126 @@
+		Args: cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			hostName := args[0]
 			cert, err := verifyCoordinator(hostName, eraConfig, insecureEra)
 			if err != nil {
 				return err
 			}
-			fmt.Println("Successfully verified coordinator, now requesting manifest signature")
-			response, err := cliDataGet(hostName, "manifest", "data.ManifestSignature", cert)
+			fmt.Println("Successfully verified coordinator, now requesting manifest")
+			response, err := cliDataGet(hostName, "manifest", "data", cert)
 			if err != nil {
 				return err
 			}
+			manifest, err := decodeManifest(consolidate, gjson.GetBytes(response, "Manifest").String(), hostName, cert)
+			if signature {
+				// wrap the signature and manifest into one json object
+				manifest = fmt.Sprintf("{\n\"ManifestSignature\": \"%s\",\n\"Manifest\": %s}", gjson.GetBytes(response, "ManifestSignature"), manifest)
+			}
+
 			if len(output) > 0 {
-				return ioutil.WriteFile(output, response, 0644)
+				return ioutil.WriteFile(output, []byte(manifest), 0644)
 			}
-			fmt.Printf("Manifest signature: %s\n", string(response))
+			fmt.Println(manifest)
 			return nil
 		},
 		SilenceUsage: true,
 	}
-	cmd.Flags().StringVarP(&output, "output", "o", "", "Save singature to file instead of printing to stdout")
+
+	cmd.Flags().BoolVarP(&signature, "signature", "s", false, "Set to additionally display the manifests signature")
+	cmd.Flags().BoolVarP(&consolidate, "display-update", "u", false, "Set to merge updates into the displayed manifest")
+	cmd.Flags().StringVarP(&output, "output", "o", "", "Save output to file instead of printing to stdout")
 	return cmd
 }
+
+// decodeManifest parses a base64 encoded manifest and optionally merges updates
+func decodeManifest(consolidate bool, encodedManifest, hostName string, cert []*pem.Block) (string, error) {
+	manifest, err := base64.StdEncoding.DecodeString(encodedManifest)
+	if err != nil {
+		return "", err
+	}
+
+	if !consolidate {
+		return string(manifest), nil
+	}
+
+	log, err := cliDataGet(hostName, "update", "data", cert)
+	if err != nil {
+		return "", err
+	}
+
+	return consolidateManifest(manifest, log)
+}
+
+// consolidateManifest updates a base manifest with values from an update log
+func consolidateManifest(rawManifest, log []byte) (string, error) {
+	var baseManifest manifest.Manifest
+	if err := json.Unmarshal(rawManifest, &baseManifest); err != nil {
+		return "", err
+	}
+
+	pkg := gjson.GetBytes(log, "..#.package").Array()
+	svn := gjson.GetBytes(log, "..#.new version").Array()
+	for idx, sPkg := range pkg {
+		*baseManifest.Packages[sPkg.String()].SecurityVersion = uint(svn[idx].Uint())
+	}
+
+	updated, err := json.Marshal(baseManifest)
+	if err != nil {
+		return "", err
+	}
+	var removeTarget map[string]interface{}
+	if err := json.Unmarshal(updated, &removeTarget); err != nil {
+		return "", err
+	}
+	removeNil(removeTarget)
```

Comment:
Why are we doing this?

Reply:
Prevents all the unset values from being displayed:

Output from `manifest get $MARBLERUN -u` with nil removal:
```javascript
{
    "Marbles": {
        "hello": {
            "MaxActivations": 0,
            "Package": "world",
            "Parameters": {
                "Argv": ["foo", "bar"]
            }
        }
    },
    "Packages": {
        "world": {
            "Debug": true,
            "SignerID": "",
            "UniqueID": "e7a6a1f81bfce87067d8f6e2aada15a1053b233fd87fd533b13492617a40646a"
        }
    }
}
```

and without:
```javascript
{
    "Packages": {
        "world": {
            "Debug": true,
            "UniqueID": "e7a6a1f81bfce87067d8f6e2aada15a1053b233fd87fd533b13492617a40646a",
            "SignerID": "",
            "ProductID": null,
            "SecurityVersion": null
        }
    },
    "Infrastructures": null,
    "Marbles": {
        "hello": {
            "Package": "world",
            "MaxActivations": 0,
            "Parameters": {
                "Argv": ["foo", "bar"]
            },
            "TLS": null
        }
    },
    "Users": null,
    "Clients": null,
    "Secrets": null,
    "RecoveryKeys": null,
    "Roles": null,
    "TLS": null
}
```